terminus/builders/osm_city_builder.py

```diff
@@ -210,3 +251,19 @@ def _translate_coords(self, lat, lon):
         coords = LatLon(lat, lon)
         (delta_lat, delta_lon) = self.map_origin.delta_in_meters(coords)
         return Point(delta_lon, delta_lat, 0)
+
+    def _check_intersection_with_bounding_box(self, prev_coord, coord):
+        '''
+        Checks if a road segment (provided 2 coordinates for the nodes)
+        intersects the bounding box
+        '''
+        segment = LineSegment(prev_coord, coord)
+
+        # Check if segment intersects with bounding box
+        intersections = [
```

Comment:
Why is `find_intersection` a class method and not an instance one? I was expecting something like:

```
intersections = [
            segment.find_intersection(self.bounds['top']),
            segment.find_intersection(self.bounds['right']),
            segment.find_intersection(self.bounds['bottom']),
            segment.find_intersection(self.bounds['left'])
]
```
but maybe I'm missing something.